git_hooks/pre-push

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+# Called by "git push" after it has checked the remote status, but before anything has been
+# pushed.  If this script exits with a non-zero status nothing will be pushed.
+#
+# This hook is called with the following parameters:
+#
+# $1 -- Name of the remote to which the push is being done
+# $2 -- URL to which the push is being done
+#
+# If pushing without using a named remote those arguments will be equal.
+#
+# Information about the commits which are being pushed is supplied as lines to
+# the standard input in the form:
+#
+#   <local ref> <local sha1> <remote ref> <remote sha1>
+#
+
+remote="$1"
+url="$2"
+
+echo "remote is $remote"
+echo "url is $url"
+
+./tests/scripts/check-doxy-blocks.pl
+if [ $? != 0 ]
+then
+exit 1
+fi
+
+./tests/scripts/check-names.sh
+if [ $? != 0 ]
+then
+exit 1
+fi
+
+./tests/scripts/check-generated-files.sh
+if [ $? != 0 ]
+then
+exit 1
+fi
```

Comment:
Actually, when I will add the set -eu in the script, I don't think we need to add the check function. and no need to check the return code of the scripts